test/commands/auth/jwt/grant.nut.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from 'path';
+import { execCmd, TestSession, prepareForJwt } from '@salesforce/cli-plugins-testkit';
+import { expect } from 'chai';
+import { Env } from '@salesforce/kit';
+import { ensureString, getString } from '@salesforce/ts-types';
+import { AuthFields } from '@salesforce/core';
+import { Result, expectPropsToExist, scrubSecrets } from '../../../testHelper';
+
+let testSession: TestSession;
+
+describe('auth:jwt:grant NUTs', () => {
+  const env = new Env();
+  let jwtKey: string;
+  let username: string;
+  let instanceUrl: string;
+  let clientId: string;
+
+  before('prepare session and ensure environment variables', () => {
+    username = ensureString(env.getString('TESTKIT_HUB_USERNAME'));
+    instanceUrl = ensureString(env.getString('TESTKIT_HUB_INSTANCE'));
+    clientId = ensureString(env.getString('TESTKIT_JWT_CLIENT_ID'));
+    ensureString(env.getString('TESTKIT_JWT_KEY'));
+
+    testSession = TestSession.create({ authStrategy: 'NONE' });
+    jwtKey = prepareForJwt(testSession.homeDir);
+  });
+
+  after(async () => {
+    await testSession.clean();
+  });
+
+  afterEach(() => {
+    execCmd(`auth:logout -p -u ${username}`, { ensureExitCode: 0 });
+  });
+
+  it('should authorize an org using jwt (json)', () => {
+    const command = `auth:jwt:grant -d -u ${username} -i ${clientId} -f ${jwtKey} -r ${instanceUrl} --json`;
+    const json = execCmd(command, { ensureExitCode: 0 }).jsonOutput as Result<AuthFields>;
+
+    expectPropsToExist(json.result, 'accessToken');
+    const auths = scrubSecrets(json.result);
+    expect(auths).to.deep.equal({
+      username,
+      orgId: '00DB0000000EfT0MAK',
+      loginUrl: 'https://login.salesforce.com/',
+      privateKey: path.join(testSession.homeDir, 'jwtKey'),
+      instanceUrl: 'https://gs0-dev-hub.my.salesforce.com',
```

Comment:
Same with this. What if we want to target this nut on a prod instance?

Reply:
I agree it is a little weird to have asserted values in env vars, but hardcoded doesn't seem right either. You could create a core connection, get the instance URL, and query for the org ID before asserting. I kinda like that solution because it validates a valid token after the exchange.

Reply:
@amphro what if we didn't check the values and instead just checked that those properties existed? After all, plugin-auth isn't responsible for creating the auth info

Reply:
ex: https://github.com/salesforcecli/plugin-user/blob/7cf495ec31a3ce847d38b14ea8c93b8167e37397/test/allCommands.nut.ts#L30

Reply:
That's fine with me. I just don't want it hardcoded.